src/reason-parser/reason_parser.mly

```diff
@@ -3710,7 +3762,24 @@ mark_position_pat
 pattern_optional_constraint:
 mark_position_pat
   ( pattern                 { $1 }
-  | pattern COLON core_type { mkpat(Ppat_constraint($1, $3)) }
+  | pattern COLON core_type
+    { mkpat(Ppat_constraint($1, $3)) }
+  (* If we kill the `let module …` syntax, this can be placed inside pattern.
```

Comment:
So apparently @jaredly still uses `let module`. Otherwise I think we can kill `let module`?